client/asset/dcr/dcr.go

```diff
@@ -1391,67 +1388,52 @@ type compositeUTXO struct {
 	var sum uint64
 	utxos := make([]*compositeUTXO, 0, len(unspents))
 	for _, txout := range unspents {
-		if txout.Confirmations >= int64(confs) {
-			scriptPK, err := hex.DecodeString(txout.ScriptPubKey)
-			if err != nil {
-				return nil, 0, fmt.Errorf("error decoding pubkey script for %s, script = %s: %v", txout.TxID, txout.ScriptPubKey, err)
-			}
-			redeemScript, err := hex.DecodeString(txout.RedeemScript)
-			if err != nil {
-				return nil, 0, fmt.Errorf("error decoding redeem script for %s, script = %s: %v", txout.TxID, txout.RedeemScript, err)
-			}
+		scriptPK, err := hex.DecodeString(txout.ScriptPubKey)
+		if err != nil {
+			return nil, 0, fmt.Errorf("error decoding pubkey script for %s, script = %s: %v", txout.TxID, txout.ScriptPubKey, err)
+		}
+		redeemScript, err := hex.DecodeString(txout.RedeemScript)
+		if err != nil {
+			return nil, 0, fmt.Errorf("error decoding redeem script for %s, script = %s: %v", txout.TxID, txout.RedeemScript, err)
+		}
 
-			nfo, err := dexdcr.InputInfo(scriptPK, redeemScript, chainParams)
-			if err != nil {
-				if errors.Is(err, dex.UnsupportedScriptError) {
-					continue
-				}
-				return nil, 0, fmt.Errorf("error reading asset info: %v", err)
+		nfo, err := dexdcr.InputInfo(scriptPK, redeemScript, chainParams)
+		if err != nil {
+			if errors.Is(err, dex.UnsupportedScriptError) {
+				continue
 			}
-			utxos = append(utxos, &compositeUTXO{
-				rpc:   txout,
-				input: nfo,
-				confs: txout.Confirmations,
-			})
-			sum += toAtoms(txout.Amount)
+			return nil, 0, fmt.Errorf("error reading asset info: %v", err)
 		}
+		utxos = append(utxos, &compositeUTXO{
+			rpc:   txout,
+			input: nfo,
+			confs: txout.Confirmations,
+		})
+		sum += toAtoms(txout.Amount)
 	}
+	// Sort in ascending order by amount (smallest first).
+	sort.Slice(utxos, func(i, j int) bool { return utxos[i].rpc.Amount < utxos[j].rpc.Amount })
 	return utxos, sum, nil
 }
 
 // lockedAtoms is the total value of locked outputs, as locked with LockUnspent.
-// TODO: handle multiple accounts!
 func (dcr *ExchangeWallet) lockedAtoms() (uint64, error) {
-	lockedOutpoints, err := dcr.node.ListLockUnspent()
+	// Fetch locked outputs for the ExchangeWallet account using RawRequest.
+	var lockedOutpoints []chainjson.TransactionInput
+	err := dcr.nodeRawRequest(methodListLockUnspent, anylist{dcr.acct}, &lockedOutpoints)
 	if err != nil {
 		return 0, err
 	}
 	var sum uint64
-	dcr.fundingMtx.Lock()
-	defer dcr.fundingMtx.Unlock()
-	for _, outPoint := range lockedOutpoints {
-		opID := outpointID(&outPoint.Hash, outPoint.Index)
-		utxo, found := dcr.fundingCoins[opID]
-		if found {
-			sum += utxo.op.value
-			continue
-		}
-		txOut, err := dcr.node.GetTxOut(&outPoint.Hash, outPoint.Index, true)
-		if err != nil {
-			return 0, err
-		}
-		if txOut == nil {
-			// Must be spent now?
-			dcr.log.Debugf("ignoring output from listlockunspent that wasn't found with gettxout. %s", opID)
-			continue
```

Comment:
There was a dcrwallet fix that fixed this issue, correct?  Can you please call that out in github comments and a commit message to justify change?  Just not that git commit messages must refer to commit hashes, but not github links.

Reply:
Right. Went back to review the related discussion in #471 and #453 and maybe I should bring back the `gettxout` check, especially because of your [last comment on 471](https://github.com/decred/dcrdex/issues/471#issuecomment-655886321).

It seems dcrwallet's `listlockunspent` still returns coins that are already spent, although that should have been fixed. If that's the case, calling `gettxout` for each coin returned from `listlockunspent` helps to prevent including a spent output in the locked balance calculation.

Worth noting again though, especially as per [this comment on 453](https://github.com/decred/dcrdex/issues/453#issuecomment-641366790), `gettxout` may not identify spent outputs if the spending tx is still in mempool, so even with `gettxout`, we might still have cases where `listlockunspent` returns a spent output and `gettxout` does not call this out. If `gettxout` does return nil for such output, we'd know it's spent and ignore it.

Reply:
I no longer see the issue in https://github.com/decred/dcrdex/issues/471.  Has been a while actually.

@buck54321 may need to weight in here as the new `logSplitFunds` probably enters into this.

Reply:
I think it's fixed via https://github.com/decred/dcrwallet/pull/1753, but I'm nervous about a regression.  Let's go with it as-is here and see if we run into any trouble